lib/graphql/analysis/ast/query_complexity.rb

```diff
@@ -21,13 +21,16 @@ class ScopedTypeComplexity
           # since the lexical binding isn't important.
           HASH_CHILDREN = ->(h, k) { h[k] = {} }
 
+          attr_reader :field_definition, :response_path, :query
+
           # @param node [Language::Nodes::Field] The AST node; used for providing argument values when necessary
           # @param field_definition [GraphQL::Field, GraphQL::Schema::Field] Used for getting the `.complexity` configuration
           # @param query [GrpahQL::Query] Used for `query.possible_types`
-          def initialize(node, field_definition, query)
+          def initialize(node, field_definition, query, response_path)
+            @node = node
             @field_definition = field_definition
             @query = query
-            @node = node
+            @response_path = response_path
```

Comment:
This is the only thing specific to our use case but I can't find a better way to generalize it